colrev/ops/pages/synthesizedrecords.py

```diff
@@ -0,0 +1,259 @@
+#! /usr/bin/env python3
+"""dashboard table and graphs for synthesized records"""
+from __future__ import annotations
+
+import bibtexparser
+import dash
+import pandas as pd
+import plotly.express as px
+import plotly.graph_objects as go
+from dash import callback
+from dash import dash_table
+from dash import dcc
+from dash import html
+from dash import Input
+from dash import Output
+
+dash.register_page(__name__)
+
+
+with open(
+    "./data/records.bib"
+) as bibtex_file:  # changing file format to csv for pandas
+    bib_database = bibtexparser.load(bibtex_file)
+    if not bib_database.entries:  # checking if bib_database file is empty
+        raise Exception("Die Datei 'records.bib' ist leer.")  # throwing Exception
+df = pd.DataFrame(bib_database.entries)
+df.to_csv("./data/records.csv", index=True)
+data = pd.read_csv("./data/records.csv").query("colrev_status == 'rev_synthesized'")
+data.rename(columns={"Unnamed: 0": "index"}, inplace=True)
+
+for title in data:
+    if (
+        title != "title"
+        and title != "author"
+        and title != "year"
+        and title != "journal"
+    ):
+        data.pop(title)
+
+
+def empty_figure() -> object:
+    """creates an empty figure in case of invalid search"""
+    figure = go.Figure(go.Scatter(x=[], y=[]))
+    figure.update_layout(template=None)
+    figure.update_xaxes(showgrid=False, showticklabels=False, zeroline=False)
+    figure.update_yaxes(showgrid=False, showticklabels=False, zeroline=False)
+
+    return figure
+
+
+def visualization_time(data) -> object:
+    """creates graph about papers published over time"""
+    if data.empty:
+        return empty_figure()
+    data2 = data.groupby(["year"])["year"].count().reset_index(name="count")
+    fig = px.bar(
+        data2,
+        x="year",
+        y="count",
+        template="simple_white",
+        title="Profile of papers published over time",
+        color="year",
+    )
+    fig.update_traces(marker_color="#fcb61a")
+
+    fig.update_layout(
+        title=dict(
+            text="<b>Profile of papers published over time</b>",
+            font=dict(family="Lato, sans-serif", size=30),
+            automargin=True,
+            x=0.5,
+        )
+    )
+    fig.update_xaxes(
+        title_text="Year",
+        type="category",
+        title_font=dict(family="Lato, sans-serif", size=20),
+        tickfont=dict(family="Lato, sans-serif", size=20),
+    )
+    fig.update_yaxes(
+        title_text="Count",
+        title_font=dict(family="Lato, sans-serif", size=20),
+        tickfont=dict(family="Lato, sans-serif", size=20),
+    )
+    return fig
+
+
+def visualization_magazines(data) -> px.bar:
+    """creates graph about papers published per journal"""
+    if data.empty:
+        return empty_figure()
+    data2 = data.groupby(["journal"])["journal"].count().reset_index(name="count")
+    fig = px.bar(
+        data2,
+        x="journal",
```

Comment:
Switching x and y could be useful to maintain readability when there are many journals.